__tests__/server/utils/onModuleLoad.spec.jsx

```diff
@@ -443,21 +444,6 @@ describe('onModuleLoad', () => {
     expect(setConfigureRequestLog).toHaveBeenCalledWith(configureRequestLog);
   });
   it('Throws error if csp and ONE_DANGER_DISABLE_CSP is not set', () => {
-    expect(() => cspCheck(missingCsp)).toThrow('Root module must provide a valid content security policy. If you would like to bypass this, set the ONE_DANGER_DISABLE_CSP enviornment variable to *, and NODE_ENV to development.');
-  });
-  it('Throws error if ONE_DANGER_DISABLE_CSP is set, and NODE_ENV is not in development', () => {
-    process.env.ONE_DANGER_DISABLE_CSP = '*';
-    delete process.env.NODE_ENV;
-    expect(() => cspCheck(missingCsp)).toThrow('If you are trying to bypass csp requirment, NODE_ENV must also be set to development.');
-    process.env.NODE_ENV = 'development';
-    delete process.env.ONE_DANGER_DISABLE_CSP;
-  });
-  it('Warns that csp header will not be set because ONE_DANGER_DISABLE_CSP is present and NODE_ENV is in development', () => {
-    process.env.ONE_DANGER_DISABLE_CSP = '*';
-    process.env.NODE_ENV = 'development';
-    cspCheck(missingCsp);
-    expect(consoleWarnSpy).toHaveBeenCalledTimes(1);
-    delete process.env.ONE_DANGER_DISABLE_CSP;
-    delete process.env.NODE_ENV;
+    expect(() => validateCspIsPresent(missingCsp)).toThrow('Root module must provide a valid content security policy.');
```

Comment:
Can we include assertions for the other cases?

- CSP is set
- CSP is not set and the env var is

Reply:
well this gets tested in csp.spec.js as such 
```js it('sets a csp header in development', () => {
      process.env.NODE_ENV = 'development';
      const cspMiddleware = requireCSP().default;
      cspMiddleware()(req, res, next);
      // eslint-disable-next-line no-underscore-dangle
      const headers = res._getHeaders();
      expect(headers).toHaveProperty('content-security-policy');
      expect(headers).not.toHaveProperty('content-security-policy-report-only');
    });

    it('does not set csp header if ONE_DANGEROUSLY_DISABLE_CSP is present', () => {
      process.env.ONE_DANGEROUSLY_DISABLE_CSP = 'true';
      const cspMiddleware = requireCSP().default;
      cspMiddleware()(req, res, next);
      // eslint-disable-next-line no-underscore-dangle
      const headers = res._getHeaders();
      expect(headers).not.toHaveProperty('content-security-policy');
    });
```
other files are managing the lack of csp so Im not sure what exactly I would be testing for within onModuleLoad

Reply:
They are different conditions though. One could break one and not the other.